numpy/lib/tests/test_stride_tricks.py

```diff
@@ -324,7 +325,9 @@ def test_as_strided():
     assert_equal(a.dtype, a_view.dtype)
 
     # Make sure that the only type that could fail is properly handled
-    dt = np.dtype({'names': [''], 'formats': ['V4']})
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Do you need the `w`?

Reply:
And if not, why the always? AFAICT, you just want to suppress the warning. For that see `suppress_warnings`. There is also `pytest.mark.filterwarnings`, but I don't know if we trust it.

Reply:
fixed to use suppress_warnings